# train_series_descriptions.csv

---

Nuestras carpetas están organizadas en un principio en "../data/raw/" y tenemos las carpetas:

 `test_images`, `train_images`,
 
 y los archivos:
 
  `sample_submission.csv`, `test_series_descriptions.csv`, `train_label_coordinates.csv`, `train_series_descriptions.csv` y `train.csv`.

---

### Step 1: Problem statement and data collection

**Analysis Objective**

**Context os the analysis**


In [2]:
# Import the dataset to start working with it

import pandas as pd 

df_train_series = pd.read_csv("../data/raw/train_series_descriptions.csv")
df_train_series.head(3)

study_id   series_id series_description
0   4003253   702807833   Sagittal T2/STIR
1   4003253  1054713880        Sagittal T1
2   4003253  2448190387           Axial T2

#### **Files**

**train_series_descriptions.csv**

- `study_id` -->

- `series_id` -->

- `series_description` -->

---

### Step 2: Exploration and data cleaning

- #### 2.1 Summarize DF

In [3]:
# Obtain dimensions

rows, columns = df_train_series.shape

print(f"The dimensions of this dataset are: {rows} Rows and {columns} Columns")

The dimensions of this dataset are: 6294 Rows and 3 Columns


In [4]:
# Obtain information about data types and non-null values

df_train_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6294 entries, 0 to 6293
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   study_id            6294 non-null   int64 
 1   series_id           6294 non-null   int64 
 2   series_description  6294 non-null   object
dtypes: int64(2), object(1)
memory usage: 147.6+ KB


In [5]:
# Check null values

null_var = df_train_series.isnull().sum().loc[lambda x: x > 0]

num_of_null_var = len(null_var)

print(f"{null_var}\n\nThe number of null variables are {num_of_null_var}")

Series([], dtype: int64)

The number of null variables are 0


---------**Conclusiones**---------

- There are a total of 6294 rows y 3 columns.

- The number of null variables are 0

- The data has 1 categorical characteristic and 2 numerical.

---

- #### 2.2 Eliminate duplicates

In [6]:
# Obtain the number of duplicate values in our dataset

print(f"The number of duplicate values in the dataset is {df_train_series.duplicated().sum()}")

The number of duplicate values in the dataset is 0
